text/0000-amcl-replacement.md

```diff
@@ -0,0 +1,41 @@
+# Cartographer AMCL replacement
+
+## Summary
+[summary]: #summary
+
+Make cartographer in Pure localization mode works as a replacement for AMCL and be one of the ROS localization method.
+
+## Motivation
+[motivation]: #motivation
+
+Cartographer is really powerful in term of localization and a pure localization mode was developed (it works already great). In order to be adopted by the ROS community it should be simple to run and works similarly  in its implementation as AMCL
+
+
+## Approach
+[approach]: #approach
+
+* Use parameters to initialize the starting pose of the robot, similarly as ` ~initial_pose_x`, ` ~initial_pose_y`, `~initial_pose_a` in AMCL: http://wiki.ros.org/amcl
+
+
+* Support the `/initialpose` topic to relocate the robot. 
+This topic is for instance used out of the box by rviz `2D pose estimate` button (or keyboard shortcut: p). 
+A quick and dirty implementation was done here: https://github.com/googlecartographer/cartographer_ros/pull/637
+
+
+* Publish the static map against which localization is done on the `/map` topic for the navigation stack to be able to be run on top of it (global planner for instance) and to have a static reference to record waypoint in.
+Currently with the occupancy_grid node, the map which is published is a superposition of the static map + the current submap. 
+A workaround is to not run the occupancy_grid node and in parallel run a `map_server` node which publish a *.pgm/*.png/ + *.yaml map that had to be generated from the same *.pbstream file which used by the pure localization node. 
+It would be nice to have everything work in one step.
```

Comment:
I agree that this should be done in one step. I suggest adding options to blacklist or whitelist trajectories from the occupancy_grid_node, so that only the map is published and not the pure localization trajectories.

Reply:
Adding a blacklist sounds good to me. @doisyg Can you update the RFC and officially make this the plan? (if you agree that this solves your usecase)

Reply:
Update: this "black/whitelist" feature was merged in https://github.com/googlecartographer/cartographer_ros/pull/899 (--> https://github.com/googlecartographer/cartographer_ros/commit/f5b583fde8d857f72c45da3820389b4227d3b084)

Reply:
And: another, very efficient alternative to publish a static map is the `pbstream_map_publisher` which I added in the meantime ([code](https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/cartographer_ros/pbstream_map_publisher_main.cc)). Just run:
```
rosrun cartographer_ros cartographer_pbstream_map_publisher --pbstream_filename xyz.pbstream --resolution 0.x --map_frame_id foo --map_topic bar
```